### PR TITLE
Fix Arg.all() after #30 + add more tests + small refactor

### PR DESCRIPTION
**Pull request by Karql**
Hi @ffMathy ;)

Today I've noticed that after #30 `Arg.all()` stops working when number of arguments is greater than 1.

```
--for(var i=0;i<Math.min(b.length, a.length);i++) {
++for(var i=0;i<Math.max(b.length, a.length);i++) {
```

For e.g. [Arg.all()] and [1,2] second iteration returns false.
With min is also not correct because [1,2] and [1,2,3] return true.

I've added 
```
    if (a.find(x => x instanceof AllArguments) || b.find(b => b instanceof AllArguments)) {
        return true;
    }
```
in `areArgumentArraysEqual`

Rest changes:
- more tests
- small refactor

I hope you will enjoy it ;)

Regards!

CC: @domasx2

### CODE REVIEW

**Comment by ffMathy** [comment]
Great work! I made a single comment.

**Comment by Karql** [comment]
@ffMathy I don't see any comments

**Comment by ffMathy** [comment]
Oh wow, forgot to hit "submit review".

File: spec/index.test.ts [hunk 1]
```diff
@@ -238,16 +237,6 @@ test('partial mocks using property instance mimicks', t => {
 	t.deepEqual(substitute.d, 1337);
 });
 
-test('are arguments equal', t => {
```
[the remark this answers is not in the document]
Comment (Karql) [reply spec/index.test.ts]: I've moved those tests to Utilities.spec.ts

e.g. 52 line: `t.true(areArgumentArraysEqual([Arg.any()], ['hi']));`

`areArgumentArraysEqual` use `areArgumentsEqual` so imo there is no need to test `areArgumentsEqual` separately ;)

**Comment by ffMathy** [comment]
Thank you so much for this!

**Comment by Karql** [comment]
No problem!
I like this lib so I'll try to help make it event better ;)

**Comment by ffMathy** [comment]
Very cool! It's very appreciated. Perhaps I can put you as an official collaborator when I see some more PRs :+1:

**Pull request merged**